tests/test_api.py

```diff
@@ -0,0 +1,263 @@
+        self.small.load_dataset(rest_platform)
+        backend = cast(RestBackend, rest_platform.backend)
+
+        res = backend.client.patch("meta/?table=true")
+        self.assert_table_res(
+            res,
+            no_of_rows=6,
+            has_columns=["run__id", "key", "type"],
+        )
+
+        res = backend.client.patch("meta/?table=true&join_run_index=true")
+        self.assert_table_res(
+            res,
+            no_of_rows=6,
+            has_columns=["model", "scenario", "version"],
+        )
+
+        res = backend.client.patch("meta/?table=false&join_run_index=true")
+        assert res.status_code == 400
+
+    def test_index_model(self, rest_platform: ixmp4.Platform):
+        self.small.load_dataset(rest_platform)
+        backend = cast(RestBackend, rest_platform.backend)
+        table_endpoint = "iamc/models/?table=True"
+
+        res = backend.client.patch(table_endpoint)
+        self.assert_table_res(res, no_of_rows=2, has_columns=["name"])
+
+        res = backend.client.patch(
+            table_endpoint, json={"run": {"scenario": {"name__in": ["Scenario 1"]}}}
+        )
+        self.assert_table_res(
+            res,
+            has_data_for_columns={"name": ["Model 1"]},
+        )
+
+        res = backend.client.patch(
+            table_endpoint, json={"run": {"scenario": {"name": "Scenario 2"}}}
+        )
+
+        self.assert_table_res(
+            res,
+            has_data_for_columns={"name": ["Model 2"]},
+        )
+
+        res = backend.client.patch(
+            table_endpoint, json={"variable": {"name": "Variable 1"}}
+        )
+
+        self.assert_table_res(
+            res,
+            has_data_for_columns={"name": ["Model 1"]},
+        )
+
+    def test_index_scenario(self, rest_platform: ixmp4.Platform):
+        self.small.load_dataset(rest_platform)
+        backend = cast(RestBackend, rest_platform.backend)
+        table_endpoint = "iamc/scenarios/?table=True"
+
+        res = backend.client.patch(table_endpoint)
+        self.assert_table_res(
+            res,
+            no_of_rows=2,
+            has_columns=["name"],
+        )
+
+        res = backend.client.patch(
+            table_endpoint, json={"run": {"model": {"name__in": ["Model 1"]}}}
+        )
+        self.assert_table_res(
+            res,
+            has_data_for_columns={"name": ["Scenario 1"]},
+        )
+
+        res = backend.client.patch(
+            table_endpoint, json={"run": {"model": {"name": "Model 2"}}}
+        )
+        self.assert_table_res(
+            res,
+            has_data_for_columns={"name": ["Scenario 2"]},
+        )
+
+        res = backend.client.patch(
+            table_endpoint, json={"variable": {"name": "Variable 4"}}
+        )
+        self.assert_table_res(
+            res,
+            has_data_for_columns={"name": ["Scenario 1", "Scenario 2"]},
+        )
+
+    def test_index_region(self, rest_platform: ixmp4.Platform):
```

Comment:
Should similar tests be added for the optimization items (in a follow-up PR)? Or do we still test their api layers with the new setup?